Scripts/GetAutoShutdownPolicyStatus/GetAutoShutdownPolicyStatus.ps1

```diff
@@ -0,0 +1,98 @@
+﻿# Example ./GetAutoShutdownPolicyStatus.ps1 -SubscriptionIds ('sub1','sub2','sub3')
+param
+(
+    [Parameter(Mandatory = $false, HelpMessage = "If you want to optionally target specific subscriptions instead of all of them, pass them in as a set")]
+    [Array] $SubscriptionIds,
+
+    [Parameter(Mandatory = $false, HelpMessage = "If you want to optionally target a specific lab instead of all of them, pass the name in")]
+    [String] $LabName
+
+)
+
+$console = (Get-Host).PrivateData
+$console.VerboseForegroundColor = "Gray"
+$console.WarningForegroundColor = "Yellow"
+
+if ($SubscriptionIds -eq $null) {
+    $SubscriptionIds = Get-AzureRmSubscription
+}
+
+foreach ($subscription in $SubscriptionIds) {
+
+    # select the subscription
+    $sub = Select-AzureRmSubscription -SubscriptionId $subscription
+
+    if ($sub -eq $null) {
+        Write-Output "Unable to find any subscriptions.  Perhaps you need to run 'Add-AzureRmAccount' and login before running this script? Unable to proceed."
+        return
+    }
+
+    # Give me all labs in the subscription
+    if ($LabName) {
+        Write-Verbose "Getting lab $($LabName)"
+        $devTestLabs = Find-AzureRmResource -ResourceType 'Microsoft.DevTestLab/labs' -ResourceNameContains $LabName
+    }
+    else {
+        $devTestLabs = Find-AzureRmResource -ResourceType 'Microsoft.DevTestLab/labs'
+    }
+    
+    foreach ($devTestLab in $devTestLabs) {
+        Write-Output "Processing lab $($devTestLab.Name)..."
+
+        $schedule = $null
+        $labHealthy = $true
+
+        # Get the AS policy status in the lab
+        try {
+            $schedule = Get-AzureRmResource -ResourceId "$($devTestLab.ResourceId)/schedules/LabVmsShutdown" -ApiVersion 2016-05-15
+        }
+        catch {
+            Write-Warning "Lab $($devTestLab.Name) auto-shutdown policy is disabled or no longer present"
+            $labHealthy = $false
+            continue
+        }
+
+        $schedDisabled = $false
+
+        if (($schedule -eq $null) -or ($schedule.Properties.status -ne 'Enabled')) {
+            Write-Warning "Lab $($devTestLab.Name) auto-shutdown policy is disabled"
+            $schedDisabled = $true
+            $labHealthy = $false
+        }
+
+        if (-not $schedDisabled) {
+            # Get all the VMs in the lab for an override
+            $virtualMachines = Get-AzureRmResource -ResourceId "$($devTestLab.ResourceId)/virtualmachines" -ApiVersion 2016-05-15 
```

Comment:
Strong type the variable to ensure it is an array even when only one record is returned, as in
```powershell
[Array] $virtualMachines = ...
```